### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by lassemand**
## Purpose

Currently the documentation does not correlate with the product, if one is to plainly follow the guide. Meaning it is not explicitly stated that certain variables has to be set in order to execute `cargo run --`.

## Changes

Add documentation about required variables which has to be set in order to start the node

### CODE REVIEW

File: concordium-node/README.md [hunk 1]
```diff
@@ -97,18 +97,20 @@ Environment variables only apply to the default build. This links with shared Ha
...
 
   ```console
-  cargo run --release --
+  cargo run --release -- --config-dir <CONCORDIUM_NODE_CONFIG_DIR> --data-dir <CONCORDIUM_NODE_DATA_DIR>
   ```
 
   to be run in release mode for improved performance. Note that the
   [concordium-consensus](../concordium-consensus/) dependency is the same regardless of how the
-  node itself is built, the `--release` only applies to the optimization of the Rust node xcomponents.
+  node itself is built, the `--release` only applies to the optimization of the rust node xcomponents.
```
Comment (abizjak) [inline concordium-node/README.md:111]: Well you could fix xcomponents as well.

File: concordium-node/README.md [hunk 2]
```diff
@@ -125,7 +127,7 @@ extra-include-dirs:
 - /opt/homebrew/include/
 ```
 
-### M1 MacOS Specific
+### Apple silicon specific
```
Comment (abizjak) [inline concordium-node/README.md:130]: This is less clear.

**Review by abizjak — CHANGES_REQUESTED** [review]

File: concordium-node/README.md [hunk 3]
```diff
@@ -97,18 +97,20 @@ Environment variables only apply to the default build. This links with shared Ha
...
-  cargo run --release --
+  cargo run --release -- --config-dir <CONCORDIUM_NODE_CONFIG_DIR> --data-dir <CONCORDIUM_NODE_DATA_DIR>
   ```
 
   to be run in release mode for improved performance. Note that the
   [concordium-consensus](../concordium-consensus/) dependency is the same regardless of how the
-  node itself is built, the `--release` only applies to the optimization of the Rust node xcomponents.
+  node itself is built, the `--release` only applies to the optimization of the rust node xcomponents.
+  
+Documentation about what <CONCORDIUM_NODE_CONFIG_DIR> and <CONCORDIUM_NODE_DATA_DIR> is seen at https://github.com/Concordium/concordium-node/blob/main/scripts/distribution/ubuntu-packages/README.md.
```
Comment (abizjak) [inline concordium-node/README.md:113]: This is not the place to refer to. This is https://github.com/Concordium/concordium-node/blob/main/VARIABLES.md#common

File: concordium-node/README.md (hunk 2 above)
Comment (lassemand) [reply concordium-node/README.md:130]: It is however more correct. Would you rather have something like M1/M2 MacOS Specific?

File: concordium-node/README.md (hunk 1 above)
Comment (lassemand) [reply concordium-node/README.md:111]: Sorry I do not understand that comment

File: concordium-node/README.md (hunk 2 above)
Comment (lassemand) [reply concordium-node/README.md:130]: I would much rather keep it as documented in this PR. The generalised term for the individual processors are indeed Apple silicon, and it is just annoying to repeatedly add individual product names.

Right now my chip (M2) does not go in under this header however it should.

Furthermore the term M1 MacOS is simply invalid and does not exist. My MacOS is Ventura while my chip is an M2.

File: concordium-node/README.md (hunk 1 above)
Comment (abizjak) [reply concordium-node/README.md:111]: `xcomponents` should be `components`. It's in the same line.

File: concordium-node/README.md (hunk 2 above)
Comment (abizjak) [reply concordium-node/README.md:130]: The instructions are related to the operating system + CPU together. If I ran linux on M1 they would not apply.

File: concordium-node/README.md (hunk 1 above)
Comment (lassemand) [inline concordium-node/README.md:111]:
```suggestion
  node itself is built, the `--release` only applies to the optimization of the rust node components.
```

File: concordium-node/README.md (hunk 2 above)
Comment (lassemand) [reply concordium-node/README.md:130]: Okay based on how the statement is specified it sounds a lot like the MacOS is M1. 

_Apple silicon MacOS specific_ or _M1/M2 MacOS specific_? or would you rather we enter each individual processor in the title?

File: concordium-node/README.md [hunk 4]
```diff
@@ -125,7 +127,7 @@ extra-include-dirs:
 - /opt/homebrew/include/
 ```
 
-### M1 MacOS Specific
+### Apple silicon + MacOS specific
```
[the remark this answers is not in the document]
Comment (lassemand) [reply concordium-node/README.md:130]:
```suggestion
### M1 and M2 MacOS specific
```

[the remark this answers is not in the document]
Comment (lassemand) [reply concordium-node/README.md:130]:
```suggestion
### M1 and M2 MacOS specific
```

**Review by abizjak — APPROVED** [review]

**Pull request merged**